src/libpriv/rpmostree-core.c

```diff
@@ -753,6 +753,12 @@ rpmostree_context_setup (RpmOstreeContext    *self,
   /* This is what we use as default. */
   dnf_context_set_rpm_macro (self->dnfctx, "_dbpath", "/" RPMOSTREE_RPMDB_LOCATION);
 
+  // Default for now for compat reasons
+  const char *db_backend = "bdb";
+  if (self->treefile_rs && ror_treefile_get_sqlite (self->treefile_rs))
+    db_backend = "sqlite";
+  dnf_context_set_rpm_macro (self->dnfctx, "_db_backend", db_backend);
```

Comment:
Hmm, but actually shouldn't this be something like:

```suggestion
  if (self->treefile_rs)
    dnf_context_set_rpm_macro (self->dnfctx, "_db_backend", ror_treefile_get_rpmdb (self->treefile_rs) ?: "bdb");
```

On the client side, we don't want to touch this macro at all, right?

Reply:
Thinking more on this, I think what we want to do is also drop a file in `/usr/lib/rpm/macros.d` so it's part of the commit itself. So then `dnf_context_set_rpm_macro` would only be used server-side to make sure the baked rpmdb type matches the macro we write in postprocessing.

Reply:
> Thinking more on this, I think what we want to do is also drop a file in /usr/lib/rpm/macros.d so it's part of the commit itself.

Hmm.  But that also will be picked up by any kind of "yum --installroot" that happens to run from the host context.  I believe we'll transparently inherit the "base commit" default because rpm will autodetect the database type in the target root.  I think.  But we clearly need to test that.

Reply:
RPM does do this correctly, as long as there's only one type of rpmdb in the rpmdb path. If there are two different types and one is specified in configuration, that one wins.